applications/FluidDynamicsApplication/custom_processes/two_fluids_inlet_process.cpp

```diff
@@ -0,0 +1,243 @@
+    mInterfacePoint = rParameters["point_on_interface"].GetVector();
+    mInletRadius = rParameters["inlet_transition_radius"].GetDouble();
+
+    // normalization of itnerface normal vector
+    if ( norm_2( mInterfaceNormal ) > 1.0e-7 ){
+        mInterfaceNormal = mInterfacePoint / norm_2( mInterfaceNormal );
+    } else {
+        KRATOS_ERROR << "Error thrown in TwoFluidsInletProcess: 'interface_normal' in 'interface_settings' must not have a norm of 0.0." << std::endl;
+    }
+    // checking to avoid 0 transition radius (necessary for division)
+    if ( mInletRadius < 1.0e-7 ){
+        KRATOS_ERROR << "Error thrown in TwoFluidsInletProcess: 'inlet_transition_radius' in 'interface_settings' must not have a value smaller or equal 0.0." << std::endl;
+    }
+
+    // setting flags for the inlet on nodes and conditions
+    #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( mrInletModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_nodes = mrInletModelPart.NodesBegin() + i_node;
+        it_nodes->Set( INLET, true );
+    }
+    #pragma omp parallel for
+    for (int i_cond = 0; i_cond < static_cast<int>( mrInletModelPart.NumberOfConditions() ); ++i_cond){
+        // iteration over all conditions
+        auto it_cond = mrInletModelPart.ConditionsBegin() + i_cond;
+        it_cond->Set( INLET, true );
+    }
+
+    // Comment: The historical DISTANCE variable is used to compute a distance field that is then stored in a non-historical variable AUX_DISTANCE
+    //          The functions for the distance computation do not work on non-historical variables - this is reason for the following procedure (*)
+
+    // (*) temporally storing the distance field as an older version of itself (it can be assured that nothing is over-written at the start)
+    #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( rRootModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_node = rRootModelPart.NodesBegin() + i_node;
+        it_node->GetSolutionStepValue(DISTANCE, 2) = it_node->GetSolutionStepValue(DISTANCE, 0);
+    }
+
+    // Preparation for variational distance calculation process
+    // setting distance of inlet nodes to 0.0
+    // setting rest of distances to 1.0
+    #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( rRootModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_node = rRootModelPart.NodesBegin() + i_node;
+        it_node->GetSolutionStepValue(DISTANCE, 0) = 1.0;
+    }
+
+    #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( mrInletModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_node = mrInletModelPart.NodesBegin() + i_node;
+        it_node->GetSolutionStepValue(DISTANCE, 0) = -1.0e-7;
+    }
+
```

Comment:
Why do you firstly set the distance to 1.0 an then to -1.0e-7?

Reply:
In this case, different model parts are concerned: RootModelPart and InletModelPart

Reply:
My apologies, I didn't note that. Resolved then.